module_7/demoblaze_tests/test_demoblaze.py

```diff
@@ -0,0 +1,48 @@
+import time
+from demoblaze_tests.common.PAGES.main_page import MainPage
+from demoblaze_tests.common.PAGES.category_page import CategoryPage
+from demoblaze_tests.common.PAGES.product_page import ProductPage
+from demoblaze_tests.common.PAGES.cart_page import CartPage
+from demoblaze_tests.common.constants import BASE_URL, CATEGORIES
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from selenium.webdriver import Chrome
+
+
+def test_log_in_for_registered_user(
+    user_creds: tuple[str, str], test_browser: "Chrome"
+):
+    """Test verifies that registered user can log in.
+    Arguments:
+    user_creds - login and password of registered user
+    test_browser - new browser window"""
+    page = MainPage(test_browser, BASE_URL)
+    page.open()
+    page.login_user(user_creds)
+    page.should_be_logged_in(user_creds[0])
+    page.logout_user()
+
+
+def test_logged_in_user_adds_product_to_cart(logged_in_user_page: MainPage):
+    """Test verifies that loggen in user can add product to cart.
+    Arguments:
+    logged_in_user_page - browser window where user already logged in"""
+    page = logged_in_user_page
+    page.go_to_category(CATEGORIES[-1])
+    monitors_page = CategoryPage(page.browser, page.browser.current_url)
```

Comment:
I think using Chaining in Python would work very well here, but it's a more advanced level.